### ISSUE TITLE
How to modify a .json file permanently ?

### ISSUE THREAD

**Issue by Skyloq** [Thutmose/Pokecube#289]
#### What happens:

Hi, I would like to edit a file in .json (Pokemobs_.json) without it then be remodeled at the start of minecraft

#### What you expected to happen:

I was expecting the file not to be reworked at the minecraft startup

#### Steps to reproduce:

1. Modifie Pokemobs_.json ( with SublimeText3 in my case ) ( for example the name of a Pokemon )
2. Save the modification
3. Launch Minecraft ( With Pokecube )
4. Back to Pokemobs_.json and see that the file has been modified ( Bulbasaur -> Bulbizarre -> Bulbasaur )
...

____
#### Affected Versions (Do *not* use "latest"):

- Pokecube Core: pokecube-core-1.12.2-6.36.12
- Pokecube Revival (if applicable): pokecube-revival-1.12.2-6.14.19
- Pokecube Mobs (if applicable): pokecube-mobs-1.12.2-4.9.0
- Minecraft: 1.12.2
- Forge: forge-1.12.2-14.23.3.2655

Thank you for reading

**Comment by Thutmose** [comment]
The default files are intended to reset each time, as they need to for any updates to the mods.  You can disable this via the forgeDatabase config in pokecube.cfg, or, you can add new json databases instead, and then update those.

in pokecube.cfg, there is the following config:
```
    S:configDatabases <
        
        
        
     >
```

if you want to edit a database, say the spawns one, and you make a copy of the file, say called pokemob_spawns_new.json, then you can apply that ingame, by the following config
```
    S:configDatabases <
        pokemob_spawns_new
        
        
     >
```

and then it will load your pokemob_spawns_new.json after loading the default ones. this allows you to overwrite the existing ones

**Comment by Skyloq** [comment]
Thank You for your answer, i will try it now

**Comment by Skyloq** [comment]
I'm probably not very talented, so I'll explain in detail what I did so you could help me

In the first place, I wrote this line of code :
![image](https://user-images.githubusercontent.com/39385087/40205617-ff37738a-5a2c-11e8-83b9-aa84a9125ff9.png)

Then I create the copy of pokemobs_spawns.json
![image](https://user-images.githubusercontent.com/39385087/40205710-6398fa06-5a2d-11e8-8391-6d3c52559013.png)

Finally, I have change "industrial" by "sandy"

![image](https://user-images.githubusercontent.com/39385087/40205749-955b82ca-5a2d-11e8-8b1c-740abf29d23a.png)
![image](https://user-images.githubusercontent.com/39385087/40205764-a4bf10a6-5a2d-11e8-9814-01c3a710c921.png)

Then my game crash :/

Can you tell me what I did wrong?

**Comment by Thutmose** [comment]
you need to send the crash report if your game crashed

**Comment by Thutmose** [comment]
also send the entire json you made, maybe something in it is formatted wrongly

**Comment by Skyloq** [comment]
There is not... Minecraft goes off and that's all. 
No return to the Launcher :/

But the loading stuck for several seconds at this time :

![image](https://user-images.githubusercontent.com/39385087/40206690-73add124-5a31-11e8-93b2-60c882e05bf4.png)

**Comment by Thutmose** [comment]
in pokecube.cfg, set debug to true, then start minecraft, then send the pokecube.log file it makes after minecraft goes off

**Comment by Skyloq** [comment]
[pokecube.zip](https://github.com/Thutmose/Pokecube/files/2014890/pokecube.zip)

I'm doing it right away

**Comment by Skyloq** [comment]
The game crashed whilst there was a severe problem during mod loading that has caused the game to fail
Error: net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Pokecube Model Loader (pokecube_ml)

**Comment by Thutmose** [comment]
if you are only editing those two spawns, use this for the json instead:
```Json
{
  "pokemon": [
    {
      "name": "Magnemite",
      "stats": {
        "spawnRules": [
          {
            "values": {
              "types": "sandy",
              "rate": "0.7"
            }
          }
        ]
      }
    },
    {
      "name": "Voltorb",
      "stats": {
        "spawnRules": [
          {
            "values": {
              "types": "sandy",
              "rate": "0.7"
            }
          }
        ]
      }
    }
  ]
}
```
no need to include all the ones you didn't edit, it results in duplicate spawns for them

**Comment by Thutmose** [comment]
get the crash report from the folder named crash-reports

**Comment by Skyloq** [comment]
[crash-2018-05-18_00.25.53-client.txt](https://github.com/Thutmose/Pokecube/files/2014900/crash-2018-05-18_00.25.53-client.txt)

**Comment by Thutmose** [comment]
hmm, I might need to look into the issues for this, in the mean time, try using the spawns.xml in config/pokecube/compat instead

**Comment by Skyloq** [comment]
I have try with the .json with two spawns and the launcher was stuck on

 "Checking entries - Magnemite"

[crash-2018-05-18_00.32.30-client.txt](https://github.com/Thutmose/Pokecube/files/2014921/crash-2018-05-18_00.32.30-client.txt)

**Comment by Thutmose** [comment]
ya,this looks like I need to fix something, in the mean time, use the spawns.xml in config/pokecube/compat and I will look into this one in the mean time

**Comment by Skyloq** [comment]
spawns.xml works great, thanks! My problem is solved but I also planned to change the name of the pokemons (to translate them into the language of my country). You think it's possible to make an xml file (same type as spawns.xml) for that?

**Comment by Thutmose** [comment]
ideally you should make a lang file for them, the ones in the database files are used internally, then translated for actual use ingame

**Comment by Thutmose** [comment]
here is the lang file with the names:
https://github.com/Thutmose/Pokecube-Mobs/tree/master/src/main/resources/assets/pokecube_mobs/lang

if you make one for your language, you can submit a pull request (PR) and I will merge it, or you can just send me the file if you don't want to make a PR

**Comment by Thutmose** [comment]
~~I have found the problem with the database files, it seems it was in https://github.com/Thutmose/Pokecube-Mobs, it will be fixed for next version of that uploaded~~

I will put the fix for this in pokecube core, I think I can get it in there.

**Comment by Thutmose** [comment]
the bug with this should be fixed in pokecube core 6.36.22

**Comment by Skyloq** [comment]
Thank you very much, I will try make a pull request when translating finished, otherwise I'll send it to you
